drivers/flash/soc_flash_nrf_mram.c

```diff
@@ -186,5 +269,31 @@ static DEVICE_API(flash, nrf_mram_api) = {
 #endif
 };
 
-DEVICE_DT_INST_DEFINE(0, NULL, NULL, NULL, NULL, POST_KERNEL, CONFIG_FLASH_INIT_PRIORITY,
-		      &nrf_mram_api);
+static int nrf_mram_init(const struct device *dev)
+{
+	struct nrf_mram_data_t *nrf_mram_data = dev->data;
+#if defined(CONFIG_NRF_IRONSIDE_BOOT_REPORT)
+	int rc = 0;
+	const struct ironside_boot_report *report;
+
+	rc = ironside_boot_report_get(&report);
+
+	if (rc) {
+		LOG_ERR("Failed to get Ironside boot report");
+		return rc;
+	}
+
+	nrf_mram_data->ironside_se_ver = FIELD_GET(IRONSIDE_SE_VER_MASK,
+						   report->ironside_se_version_int);
```

Comment:
@rghaddab Checking only this last part is not entirely forward compatible, as this number is planned to wrap around at 127.

Comparing against the full version is recommended, see https://github.com/zephyrproject-rtos/zephyr/pull/97691#discussion_r2592100564